### PR TITLE
Add Swift code generator plugin

### PR DESCRIPTION
**Pull request by timburks**
This set of commits adds a code generation plugin (written in Swift) that generates Swift code for API clients and servers for specified OpenAPI models. The plugin is not complete but intended as an illustration of a code generation plugin that could be written with Swift. This initial commit generates a generic client and a server that uses the Kitura web framework.

The plugins/go/openapi_swift_generator/examples/bookstore directory contains an example and instructions for demonstrating the plugin.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
+  - ./.travis-install.sh
 
 script:
   - go test . -v
-  - cd plugins/go/openapi_go_generator/examples/bookstore
+  - pushd plugins/go/openapi_go_generator/examples/bookstore
+  - make test
+  - popd
+  - export PATH=.:$HOME/local/bin:$PATH
+  - export LD_LIBRARY_PATH=$HOME/local/lib
```
Comment (guptasu) [inline .travis.yml]: Should this be done inside the -install.sh script ?

File: .travis-install.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
```
Comment (guptasu) [inline .travis-install.sh]: Should we put all this inside the before_install step inside the original .travis.yml file ?

File: plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/InternalLoader.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Stencil
+import Foundation
+
+// A class for loading Stencil templates from compiled-in representations
+
+public class InternalLoader: Loader {
```
Comment (guptasu) [inline plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/InternalLoader.swift]: TemplateLoader ?

File: plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/main.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  let request = try Openapi_Plugin_V1_Request(protobuf: rawRequest)
+  let wrapper = request.wrapper
+  let document = try Openapi_V2_Document(protobuf:wrapper.value)
+
+  // build the service renderer
+  let renderer = ServiceRenderer(document:document)
+
+  // generate the desired files
+  var response = Openapi_Plugin_V1_Response()
+  let filenames = ["client.swift", "service.swift", "types.swift"]
```
Comment (guptasu) [inline plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/main.swift]: Any reason why this is different than the other sample where client and server were conditionally generated ?
https://github.com/googleapis/openapi-compiler/blob/master/plugins/go/openapi_go_generator/main.go#L51

File: .travis.yml (hunk 1 above)
Comment (timburks) [inline .travis.yml]: IIRC, I tried that and the values weren't set outside of the script.

File: .travis-install.sh (hunk 2 above)
Comment (timburks) [inline .travis-install.sh]: I will try that.

File: plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/InternalLoader.swift (hunk 3 above)
Comment (timburks) [inline plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/InternalLoader.swift]: That would be fine. Here's some [background on that](https://github.com/kylef/Stencil/issues/97). Stencil has an alternate loader called FileSystemLoader that reads from the file system. I think "InternalLoader" would be a good name for use inside Stencil, but see that it's confusing in our code.

File: plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/main.swift (hunk 4 above)
Comment (timburks) [inline plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/main.swift]: Laziness. I'll add that.

Comment (timburks) [inline plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/main.swift]: Done [here](https://github.com/googleapis/openapi-compiler/pull/25/commits/a31fd29f3f6060810c6a3fd5dadff4a5b5e66d4d).

File: plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/InternalLoader.swift (hunk 3 above)
Comment (timburks) [inline plugins/swift/openapi_swift_generator/Sources/openapi_swift_generator/InternalLoader.swift]: Done [here](https://github.com/googleapis/openapi-compiler/pull/25/commits/9f9913bdac82266a55f19ec9095a57ea6c34f427).

File: .travis-install.sh (hunk 2 above)
Comment (timburks) [inline .travis-install.sh]: Can we set environment variables in the steps under install or before_install?

**Pull request merged**